Implement heterojunction in depletion approximation and fix build issues

phoebe-p opened the pull request:
- [x] add documentation for installation without PDD, either from PyPI or from local source.

iclned commented:
Build issues on Windows are fixed and heterojunction model functionality implemented for the depletion approximation.

phoebe-p committed:
Allow heterojunction in DA, update build and test systemSquashed commit of the following:

- add functionality to allow heterojunctions in the depletion approximation (i.e. do not throw an error with a heterojunction)
- :bug: fix issue where the depletion width in the depletion approximation could be wider than the relevant junction region
- :checkered_flag: :green_heart: fix Windows build in build_deploy_wheels, and fig issue where Windows test did not run but also did not throw an error in GitHub Actions
- :penguin: :apple: fix S4 build in MacOS/Linux
- add ability to install without the PDD solver (locally or from PyPI)
- :memo: update docs with instructions for the above

phoebe-p committed:
try to fix incorrect coverage issue

phoebe-p committed:
:white_check_mark: Fix/add tests and workflow

- Add test for case where depletion width > region width
- Remove testing of examples in Windows (this did not run in GitHub Actions anyway, so removing explicitly so it's clear the examples are not being run on Windows)
- Restore test for exceptions in depletion_approximation
